### PR TITLE
Add D1 vs Hyperdrive

### PR DESCRIPTION
**Pull request by thomasgauvin**

### CODE REVIEW

**Review by Oxyjun — APPROVED** [review]

**Comment by Oxyjun** [comment]
Hey @vy-ton , we wanted to run this by you - do you think it's worth surfacing any limits of D1 on this page as a factor which helps user decide between using D1 vs external database + Hyperdrive? Or would you rather not draw explicit attention to that?

File: src/content/docs/workers/platform/storage-options.mdx [hunk 1]
```diff
@@ -209,4 +209,15 @@ The new beta version of Durable Objects is available where each Durable Object h
 
 :::
 
+## D1 vs Hyperdrive
+
+D1 is a standalone, serverless database that provides a SQL API to store and access your relational data. 
+
+Hyperdrive is a service that improves the performance of your existing, regional PostgreSQL databases, optimizing them for global data access from Workers.
```
Comment (irvinebroque) [inline src/content/docs/workers/platform/storage-options.mdx]: It also allows you to connect to them in the first place

File: src/content/docs/workers/platform/storage-options.mdx [hunk 2]
```diff
@@ -209,4 +209,15 @@ The new beta version of Durable Objects is available where each Durable Object h
...
+
+If you are building a new project on Workers or are considering migrating your data, use D1.
+
+If you are building a Workers project with an existing PostgreSQL database, use Hyperdrive.
+
+D1 does not need to be used with Hyperdrive because it does not have slow connection setups which would benefit from Hyperdrive's connection pooling. D1 data can also be cached within Workers using the [Cache API](/workers/runtime-apis/cache/).
```
Comment (irvinebroque) [inline src/content/docs/workers/platform/storage-options.mdx]: Should it be more explicit that you cannot use both together?

**Comment by vy-ton** [comment]
> Hey @vy-ton , we wanted to run this by you - do you think it's worth surfacing any limits of D1 on this page as a factor which helps user decide between using D1 vs external database + Hyperdrive? Or would you rather not draw explicit attention to that?

should note D1's 10GB database limit and the preferred scale out architecture with many D1 databases

File: src/content/docs/workers/platform/storage-options.mdx [hunk 3]
```diff
@@ -209,4 +209,15 @@ The new beta version of Durable Objects is available where each Durable Object h
 
 :::
 
+## D1 vs Hyperdrive
+
+D1 is a standalone, serverless database that provides a SQL API to store and access your relational data.
```
Comment (vy-ton) [inline src/content/docs/workers/platform/storage-options.mdx]:
```suggestion
D1 is a standalone, serverless database that provides a SQL API, using SQLite's SQL semantics, to store and access your relational data. 
```

File: src/content/docs/workers/platform/storage-options.mdx (hunk 2 above)
Comment (Oxyjun) [reply src/content/docs/workers/platform/storage-options.mdx]: @thomasgauvin if it's a case where a user can't use D1 with Hyperdrive at all, I think we should state it here?

Comment (Oxyjun) [reply src/content/docs/workers/platform/storage-options.mdx]: Adding text to explicitly state you cannot use D1 with Hyperdrive.

File: src/content/docs/workers/platform/storage-options.mdx (hunk 1 above)
Comment (Oxyjun) [reply src/content/docs/workers/platform/storage-options.mdx]: Is this not already implied by the fact that Hyperdrive can improve the performance of the existing database? Happy to add it in if you think it should be explicit though.

**Review by marciocloudflare — APPROVED** [review]

File: src/content/docs/workers/platform/storage-options.mdx (hunk 1 above)
Comment (vy-ton) [reply src/content/docs/workers/platform/storage-options.mdx]: @Oxyjun Let's make it explicit. When framed as performance improvement, Hyperdrive could be seen as optional

File: src/content/docs/workers/platform/storage-options.mdx [hunk 4]
```diff
@@ -209,4 +209,20 @@ The new beta version of Durable Objects is available where each Durable Object h
 
 :::
 
+## D1 vs Hyperdrive
 
+D1 is a standalone, serverless database that provides a SQL API, using SQLite's SQL semantics, to store and access your relational data.
+
+Hyperdrive is a service that improves the performance of your existing, regional PostgreSQL databases, optimizing them for global data access from Workers.
```
Comment (vy-ton) [inline src/content/docs/workers/platform/storage-options.mdx]:
```suggestion
Hyperdrive is a service that lets you connect your existing, regional PostgreSQL databases and improves database performance by optimizing them for global data access from Workers.
```

File: src/content/docs/workers/platform/storage-options.mdx [hunk 5]
```diff
@@ -209,4 +209,20 @@ The new beta version of Durable Objects is available where each Durable Object h
...
+
+- If you are building a new project on Workers or are considering migrating your data, use D1.
+- If you are building a Workers project with an existing PostgreSQL database, use Hyperdrive.
+- If your database exceeds 10 GB (the maximum size for a D1 database), consider splitting the database to multiple, smaller D1 databases, or use your existing PostgreSQL database with Hyperdrive.
```
Comment (vy-ton) [inline src/content/docs/workers/platform/storage-options.mdx]: Combine this with the first bullet

```suggestion
- If your working data size exceeds 10 GB (the maximum size for a D1 database), consider splitting the database to multiple, smaller D1 databases.
```

**Review by vy-ton — APPROVED** [review]
@Oxyjun approved with a few modifications

File: src/content/docs/workers/platform/storage-options.mdx [hunk 6]
```diff
@@ -209,4 +209,20 @@ The new beta version of Durable Objects is available where each Durable Object h
 
 :::
 
+## D1 vs Hyperdrive
 
+D1 is a standalone, serverless database that provides a SQL API, using SQLite's SQL semantics, to store and access your relational data.
+
+Hyperdrive is a service that lets you connect your existing, regional PostgreSQL databases and improves database performance by optimizing them for global data access from Workers.
```
Comment (thomasgauvin) [inline src/content/docs/workers/platform/storage-options.mdx]:
```suggestion
Hyperdrive is a service that lets you connect to your existing, regional PostgreSQL databases and improves database performance by optimizing them for global, scalable data access from Workers.
```

File: src/content/docs/workers/platform/storage-options.mdx [hunk 7]
```diff
@@ -209,4 +209,20 @@ The new beta version of Durable Objects is available where each Durable Object h
...
+
+- If you are building a new project on Workers or are considering migrating your data, use D1.
+- If you are building a Workers project with an existing PostgreSQL database, use Hyperdrive.
+- If your working data size exceeds 10 GB (the maximum size for a D1 database), consider splitting the database to multiple, smaller D1 databases.
+
+:::caution
```
Comment (thomasgauvin) [inline src/content/docs/workers/platform/storage-options.mdx]:
```suggestion
:::note
```

File: src/content/docs/workers/platform/storage-options.mdx [hunk 8]
```diff
@@ -209,4 +209,20 @@ The new beta version of Durable Objects is available where each Durable Object h
...
 
+## D1 vs Hyperdrive
 
+D1 is a standalone, serverless database that provides a SQL API, using SQLite's SQL semantics, to store and access your relational data.
+
+Hyperdrive is a service that lets you connect your existing, regional PostgreSQL databases and improves database performance by optimizing them for global data access from Workers.
+
+- If you are building a new project on Workers or are considering migrating your data, use D1.
+- If you are building a Workers project with an existing PostgreSQL database, use Hyperdrive.
+- If your working data size exceeds 10 GB (the maximum size for a D1 database), consider splitting the database to multiple, smaller D1 databases.
```
Comment (thomasgauvin) [inline src/content/docs/workers/platform/storage-options.mdx]:
```suggestion
```

**Comment by thomasgauvin** [comment]
> > Hey @vy-ton , we wanted to run this by you - do you think it's worth surfacing any limits of D1 on this page as a factor which helps user decide between using D1 vs external database + Hyperdrive? Or would you rather not draw explicit attention to that?
> 
> should note D1's 10GB database limit and the preferred scale out architecture with many D1 databases

I don't think we should note this here. In this section, we compare D1 and Hyperdrive, and have an answer for when to use D1, when to use Hyperdrive.

**Pull request merged**